vars/runATH.groovy

```diff
@@ -0,0 +1,205 @@
+                'PATH+JAVA=${JAVA_HOME}/bin',
+                "PATH+MAVEN=${tool 'mvn'}/bin"
+
+        ]
+
+        if (!fileExists(metadataFile)) {
+            echo "Skipping ATH execution because the metadata file does not exist. Current value is ${metadataFile}."
+            skipExecution = true
+            return
+        }
+
+        stage("Getting ATH sources and Jenkins war") {
+            // Start validation
+            metadata = readYaml(file: metadataFile).ath
+            if (metadata == null) {
+                error "The provided metadata file seems invalid as it does not contain a valid ath section"
+            }
+            if (metadata == 'default') {
+                echo "Using default configuration for ATH"
+                metadata = [:]
+            } else if (metadata.browsers == null) {
+                echo "The provided metadata file does not include the browsers property, using firefox as default"
+            }
+            // Allow to override athUrl and athRevision from metadata file
+            athUrl = metadata.athUrl ?: athUrl
+            isLocalATH = athUrl.startsWith("file://")
+            athRevision = metadata.athRevision ?: athRevision
+
+            // Allow override of jenkins version from metadata file
+            jenkins = metadata.jenkins ?: jenkins
+            isVersionNumber = (jenkins =~ /^\d+([.]\d+)*$/).matches()
+
+            if (!isLocalATH) {
+                echo 'Checking connectivity to ATH sources…'
+                sh "git ls-remote --exit-code -h ${athUrl}"
+            }
+            if (jenkins == "latest") {
+                jenkinsURl = mirror + "war/latest/jenkins.war"
+            } else if (jenkins == "latest-rc") {
+                jenkinsURl = mirror + "/war-rc/latest/jenkins.war"
+            } else if (jenkins == "lts") {
+                jenkinsURl = mirror + "war-stable/latest/jenkins.war"
+            } else if (jenkins == "lts-rc") {
+                jenkinsURl = mirror + "war-stable-rc/latest/jenkins.war"
+            }
+
+            if (!isVersionNumber) {
+                echo 'Checking whether Jenkins WAR is available…'
+                sh "curl -ILf ${jenkinsURl}"
+            }
+            // Validation ended
+
+            // ATH
+            if (isLocalATH) { // Deal with already existing ATH sources
+                athSourcesFolder = athUrl - "file://"
+            } else {
+                dir(athSourcesFolder) {
+                    checkout changelog: true, poll: false, scm: [$class           : 'GitSCM', branches: [[name: athRevision]],
+                                                                 userRemoteConfigs: [[url: athUrl]]]
+                }
+            }
+            dir(athSourcesFolder) {
+                // We may need to run things in parallel later, so avoid several checkouts
+                stash name: "athSources"
+            }
+
+            // Jenkins war
+            if (isVersionNumber) {
```

Comment:
This code needs more comments as to what this code is doing. It looks like you are downloading a war, but if that is the intent please say that.